ui/src/components/inputs/FileExplorer.vue

```diff
@@ -688,21 +688,23 @@
 
     async function removeItems() {
         if(confirmation.value.nodes === undefined) return;
-        for (const node of confirmation.value.nodes) {
+        const paths = confirmation.value.nodes.map(node => filesStore.getPath(node.id) ?? "");
+        await Promise.all(confirmation.value.nodes.map(async (node, i) => {
+            const path = paths[i];
```

Comment:
Why did you need to precreate the path array? 

```suggestion
            const path = filesStore.getPath(node.id) ?? "";
```

Reply:
I was thinking to get  the path for every file from filesStore before any concurrent deletion. but yeah i get the part to have the path synchronously at the very start of the async.